## Final Assignement 

In [1]:
import pandas as pd

In [2]:
df = pd.read_csv('kc_house_data.csv')

In [4]:
df.head()

id             date     price  bedrooms  bathrooms  sqft_living  \
0  7129300520  20141013T000000  221900.0         3       1.00         1180   
1  6414100192  20141209T000000  538000.0         3       2.25         2570   
2  5631500400  20150225T000000  180000.0         2       1.00          770   
3  2487200875  20141209T000000  604000.0         4       3.00         1960   
4  1954400510  20150218T000000  510000.0         3       2.00         1680   

   sqft_lot  floors  waterfront  view  ...  grade  sqft_above  sqft_basement  \
0      5650     1.0           0     0  ...      7        1180              0   
1      7242     2.0           0     0  ...      7        2170            400   
2     10000     1.0           0     0  ...      6         770              0   
3      5000     1.0           0     0  ...      7        1050            910   
4      8080     1.0           0     0  ...      8        1680              0   

   yr_built  yr_renovated  zipcode      lat     long  sqft_living15  \
0      1955             0    98178  47.5112 -122.257           1340   
1      1951          1991    98125  47.7210 -122.319           1690   
2      1933             0    98028  47.7379 -122.233           2720   
3      1965             0    98136  47.5208 -122.393           1360   
4      1987             0    98074  47.6168 -122.045           1800   

   sqft_lot15  
0        5650  
1        7639  
2        8062  
3        5000  
4        7503  

[5 rows x 21 columns]

In [71]:
df1 = df.groupby('bedrooms')['price'].mean().reset_index()

In [72]:
df1.head()

bedrooms          price
0         0  410223.076923
1         1  317657.959799
2         2  401387.747101
3         3  466276.590900
4         4  635564.677129

**The data reveals that houses with more bedrooms tend to have higher average prices. For instance, houses with 1 bedroom show an average price of 353,042.86 while those with 2 bedrooms show an average price of 415,767.86 and this trend continues for higher numbers of bedrooms.**

In [77]:
df[['bedrooms', 'sqft_living', 'sqft_lot']].groupby('bedrooms').mean().round(1).head(4)

sqft_living  sqft_lot
bedrooms                       
0              1707.8   24141.3
1               884.6   16285.7
2              1239.8   12096.7
3              1805.8   14414.8

**The average living space and lot size increase with the number of bedrooms. This is to be expected, as houses with more bedrooms tend to be larger and have larger lots.**

In [83]:
df[['price', 'grade', 'sqft_living', 'sqft_lot']].groupby('grade').mean().round(1).reset_index()

grade      price  sqft_living  sqft_lot
0       1   142000.0        290.0   20875.0
1       3   205666.7        596.7   26953.0
2       4   214381.0        660.5   22101.5
3       5   248524.0        983.3   24019.9
4       6   301916.6       1191.6   12647.0
5       7   402593.3       1689.4   11766.4
6       8   542895.5       2184.7   13510.2
7       9   773738.2       2868.1   20638.5
8      10  1072347.5       3520.3   28191.1
9      11  1497792.4       4395.4   38372.8
10     12  2192500.0       5471.6   54521.3
11     13  3710769.2       7483.1   28204.8

**Houses with a higher grade tend to be more expensive, have more square footage, and have larger lot sizes.**

In [84]:
df[['price', 'waterfront', 'sqft_living', 'sqft_lot']].groupby('waterfront').mean().round(1).reset_index()

waterfront      price  sqft_living  sqft_lot
0           0   531653.4       2071.6   15029.0
1           1  1662524.2       3173.7   25371.8

**Waterfront and House Prices:**
- The data indicates difference in average house prices based on whether the property has a waterfront view.
- Houses with a waterfront view have higher average prices compared to those without a waterfront view. <br>
<br>

**Living Area and Lot Area:**
- We can assume that waterfront properties might have larger living and lot areas on average.
- The larger living and lot areas of waterfront properties could contribute to their higher average prices, as larger spaces are often valued more highly.

### Grade:

In [13]:
#Function to categorize grade
def categorize_grade(grade):
    if grade <= 5:
        return 'bad'
    elif grade <= 8 and grade>5:
        return 'good'
    else:
        return 'very good'

In [14]:
# Define a lambda function to categorize grade
df['grade_label'] = df['grade'].apply(lambda grade:categorize_grade(grade))
df.head()

id             date     price  bedrooms  bathrooms  sqft_living  \
0  7129300520  20141013T000000  221900.0         3       1.00         1180   
1  6414100192  20141209T000000  538000.0         3       2.25         2570   
2  5631500400  20150225T000000  180000.0         2       1.00          770   
3  2487200875  20141209T000000  604000.0         4       3.00         1960   
4  1954400510  20150218T000000  510000.0         3       2.00         1680   

   sqft_lot  floors  waterfront  view  ...  sqft_above  sqft_basement  \
0      5650     1.0           0     0  ...        1180              0   
1      7242     2.0           0     0  ...        2170            400   
2     10000     1.0           0     0  ...         770              0   
3      5000     1.0           0     0  ...        1050            910   
4      8080     1.0           0     0  ...        1680              0   

   yr_built  yr_renovated  zipcode      lat     long  sqft_living15  \
0      1955             0    98178  47.5112 -122.257           1340   
1      1951          1991    98125  47.7210 -122.319           1690   
2      1933             0    98028  47.7379 -122.233           2720   
3      1965             0    98136  47.5208 -122.393           1360   
4      1987             0    98074  47.6168 -122.045           1800   

   sqft_lot15  grade_label  
0        5650         good  
1        7639         good  
2        8062         good  
3        5000         good  
4        7503         good  

[5 rows x 22 columns]

In [15]:
df[['price', 'waterfront', 'grade','sqft_living', 'sqft_lot','grade_label']].groupby(['waterfront', 'grade_label']).mean().round(1)

price  grade  sqft_living  sqft_lot
waterfront grade_label                                         
0          bad           241756.3    4.9        944.4   23867.2
           good          437887.8    7.2       1803.7   12417.4
           very good     933310.1    9.5       3238.7   25116.1
1          bad           400725.0    5.0        817.5   21870.5
           good         1012533.9    7.4       2318.6   29122.7
           very good    2302958.3   10.2       4049.3   22189.6

- **The data suggests a positive correlation between waterfront views and house grades. This indicates that properties with a waterfront view tend to receive higher grades.**
- **Higher-grade houses tend to command higher prices, whether or not they have a waterfront view.**

In [89]:
#df[['price', 'waterfront', 'grade','sqft_living', 'sqft_lot']].groupby(['waterfront', 'grade']).mean().round(1)

price  sqft_living  sqft_lot
waterfront grade                                  
0          1       142000.0        290.0   20875.0
           3       205666.7        596.7   26953.0
           4       214381.0        660.5   22101.5
           5       245966.0        986.1   24056.0
           6       300243.5       1191.7   12600.3
           7       401476.7       1688.2   11723.1
           8       538258.9       2180.9   13387.6
           9       767489.2       2868.2   20604.5
           10     1042090.4       3516.9   28392.2
           11     1423823.6       4361.3   39418.4
           12     2036519.0       5404.2   58038.5
           13     3710769.2       7483.1   28204.8
1          5       400725.0        817.5   21870.5
           6       584382.7       1165.8   20517.4
           7       837478.3       2148.2   28633.4
           8      1241636.2       2762.5   31985.5
           9      1421140.0       2860.2   24159.2
           10     2267500.0       3652.9   20246.5
           11     2899500.0       5042.5   18558.6
           12     3312727.3       5955.3   29261.2

In [11]:
df[['price', 'waterfront', 'grade','sqft_living', 'sqft_lot','grade_label']].groupby('waterfront')['grade_label'].value_counts(normalize=True).mul(100).rename('percentage').reset_index()

waterfront grade_label  percentage
0           0        good   79.310023
1           0   very good   19.426573
2           0         bad    1.263403
3           1   very good   51.533742
4           1        good   46.012270
5           1         bad    2.453988

**Waterfront and Grade Label Distribution:**<br>
<br>
- Among non-waterfront properties (waterfront = 0), the majority (about 79.3%) fall under the 'good' grade label, followed by 'very good' grade (approximately 19.4%), and a smaller portion categorized as 'bad' grade (around 1.3%).
- For waterfront properties (waterfront = 1), the distribution is different. The highest percentage falls under the 'very good' grade label (around 51.5%), followed by 'good' grade (approximately 46%), and a smaller portion in the 'bad' grade category (about 2.5%).
<br>

**Waterfront Impact on Grade Labels:**

- Waterfront properties seem to have a higher proportion of properties labeled as 'very good' compared to non-waterfront properties. This might suggest that waterfront properties are often associated with higher quality grades..<br>
<br>

**Grade Label Distribution Variation:**
- The distribution of grade labels within the waterfront categories indicates that different property types have varying grade distributions. This could imply differences in construction quality or other property characteristics based on waterfront status.<br>

### View:

In [91]:
df[['price', 'sqft_living', 'sqft_lot', 'view']].groupby('view').mean().round(1)

price  sqft_living  sqft_lot
view                                  
0      496623.5       1997.8   14179.2
1      812518.6       2569.0   12358.9
2      792746.2       2655.3   22256.9
3      972468.4       3018.6   34788.9
4     1464362.6       3351.5   21595.0


- Houses with a better view category tend to have higher average prices.
- While the data doesn't directly show how living and lot areas relate to view categories, we can assume that houses with better views might have larger living and lot areas on average.

In [93]:
df[['price', 'sqft_living', 'sqft_lot', 'view','waterfront']].groupby(['waterfront','view']).mean().round(1)

price  sqft_living  sqft_lot
waterfront view                                  
0          0      496623.5       1997.8   14179.2
           1      813054.9       2570.9   11891.3
           2      783955.1       2649.3   22158.1
           3      964685.1       3039.2   34123.4
           4     1270713.3       3412.6   22720.3
1          1      635000.0       1940.0  167125.0
           2     1842187.5       3369.1   34060.0
           3     1173605.3       2485.3   51986.0
           4     1728299.6       3268.1   20061.3

- The data clearly shows that house with a waterfront view generally have higher average prices and different characteristics compared to houses without a waterfront view.
- Regardless of whether ahouse has a waterfront view or not, as the quality of the view category improves, the average price tends to increase. This indicates that a better view is associated with higher average prices for properties.

In [23]:
# Create a function to categorize view
def categorize_view(view):
    if view <= 2:  # Change the threshold as needed
        return 'bad'
    else:
        return 'good'

# Apply the categorize_view function to create a new 'view_label' column
df['view_label'] = df['view'].apply(categorize_view)

# Group by 'waterfront' and 'view_label', calculate the percentage, and reset the index
df[['price', 'sqft_living', 'sqft_lot', 'view','waterfront','view_label']].groupby('waterfront')['view_label'].value_counts(normalize=True).mul(100).rename('percentage').reset_index()

waterfront view_label  percentage
0           0        bad   96.853147
1           0       good    3.146853
2           1       good   94.478528
3           1        bad    5.521472

**Waterfront Impact on View Labels:**
<br>
- For non-waterfront properties (waterfront = 0), the vast majority (about 96.9%) are categorized as 'bad' view labels, while only a small portion (approximately 3.1%) fall under the 'good' view label.
- For waterfront properties (waterfront = 1), the distribution is different. A significant portion (about 94.5%) have a 'good' view label, with a smaller percentage (around 5.5%) categorized as 'bad' view labels.
<br>

**Property View Quality**
<br>
- This analysis suggests that waterfront houses tend to have a higher percentage of 'good' view labels compared to non-waterfront houses. This aligns with the general notion that waterfront properties often offer more desirable views
<br>

**Impact on Property Value:**
- There might be a correlation between the view quality and property value, particularly for waterfront properties. Properties with 'good' view labels might command higher prices compared to those with 'bad' view labels.
<br>
- We can say tat the presence of a waterfront view has a high influence on the assigned view categories.

In [25]:
df[['price', 'waterfront', 'grade',
'sqft_living', 'sqft_lot','view','grade_label','view_label']].groupby('view_label')['grade_label'].value_counts(normalize=True).mul(100).rename('percentage').reset_index()

view_label grade_label  percentage
0        bad        good   80.504234
1        bad   very good   18.196690
2        bad         bad    1.299076
3       good   very good   56.574186
4       good        good   42.822678
5       good         bad    0.603136

**View Label 'Bad':**<br>

- Within the 'bad' view label category, the majority of houses have 'good' grade labels (80.5%).
- A significant proportion also falls under the 'very good' grade label (18.2%).
- A small fraction of properties have a 'bad' grade label (1.3%).<br>
<br>

**View Label 'Good':**<br>

- For the 'good' view label category, more than half of the houses have 'very good' grade labels (56.57%).
- A substantial portion also falls under the 'good' grade label (42.82%).
- A very small fraction of properties have a 'bad' grade label (0.6%).<br>
<br>
- The analysis of the distribution of 'grade_label' within 'view_label' categories provides insights into how the quality of the view ('good' or 'bad') relates to the house grade.
- For houses with a 'bad' view label, the majority have 'good' grade labels, indicating that the quality of the view doesn't necessarily correlate with lower house grades.
- For properties with a 'good' view label, the majority have 'very good' grade labels, suggesting that houses with better views tend to have higher house grades.
- The presence of a small percentage of 'bad' grade labels in both 'bad' and 'good' view label categories could be due to other factors influencing the properthouse grades.
- Overall, this analysis emphasizes the potential correlation between view quality and house grades.

### Floors:

In [113]:
df[['price', 'sqft_living', 'sqft_lot','floors']].groupby('floors').mean().round(1)

price  sqft_living  sqft_lot
floors                                  
1.0      442219.6       1731.8   14568.4
1.5      559044.9       1907.9   16966.5
2.0      649051.5       2571.3   16133.7
2.5     1061020.9       3145.4   17347.3
3.0      582620.1       1788.3    4461.6
3.5      933937.5       2518.0    3038.8

- The analysis of mean values across different 'floors' categories provides insights into how the number of floors correlates with property attributes like price, living area size ('sqft_living'), and lot size ('sqft_lot').
- Properties with more floors tend to have higher average prices, larger living area sizes, and larger lot sizes.
- Properties with 2.5 floors have the highest average price, living area size, and lot size.
- Properties with 3.0 floors have relatively smaller lot sizes, which might be attributed to the smaller average 'sqft_lot' value.

In [27]:
df[['price', 'waterfront', 'grade_label','sqft_living', 'sqft_lot','floors']].groupby('floors')['grade_label'].value_counts(normalize=True).mul(100).rename('percentage').reset_index()

floors grade_label  percentage
0      1.0        good   91.863296
1      1.0   very good    5.955056
2      1.0         bad    2.181648
3      1.5        good   90.000000
4      1.5   very good    7.905759
5      1.5         bad    2.094241
6      2.0        good   60.611576
7      2.0   very good   39.364155
8      2.0         bad    0.024269
9      2.5   very good   56.521739
10     2.5        good   43.478261
11     3.0        good   79.119086
12     3.0   very good   20.880914
13     3.5        good   87.500000
14     3.5   very good   12.500000

- The analysis of the percentage distribution of 'grade_label' within 'floors' categories provides insights into how the number of floors in a property relates to its property grade.
- Houses with fewer floors, such as 1.0 and 1.5, tend to have higher percentages of 'good' grade labels.
- The distribution pattern shifts for houses with 2.0 floors, where a majority have 'good' grade labels and a significant portion have 'very good' grade labels.
- Houses with 2.5 floors predominantly have 'very good' grade labels, with a minority having 'good' grade labels.
- For houses with more floors, such as 3.0 and 3.5, 'good' grade labels are dominant, with a smaller percentage of 'very good' grade labels.This suggests that while properties with more floors tend to have good grades, they are less likely to receive the highest 'very good' grade labels.
- The number of floors in a house appears to have a notable correlation with its property grade.
- Properties with 2.5 floors predominantly have 'very good' grade labels, indicating that this specific number of floors might be associated with higher-quality houses.

In [118]:
df[['price', 'waterfront', 'grade',
'sqft_living', 'sqft_lot','floors']].groupby('waterfront')['floors'].value_counts(normalize=True).mul(100).rename('percentage').reset_index()

waterfront  floors  percentage
0            0     1.0   49.524476
1            0     2.0   38.069930
2            0     1.5    8.806527
3            0     3.0    2.820513
4            0     2.5    0.741259
5            0     3.5    0.037296
6            1     2.0   46.012270
7            1     1.0   34.969325
8            1     1.5   12.883436
9            1     3.0    4.907975
10           1     2.5    1.226994

**For houses without a waterfront view:**
<br>
- Most of them have 1.0 floor (49.5%), followed by 2.0 floors (38.1%).
- Smaller proportions have 1.5, 3.0, 2.5, and 3.5 floors.
<br>

**For houses with a waterfront view:**
<br>
- A significant portion has 2.0 floors (46.0%), and the second most common is 1.0 floor (35.0%).
- Some have 1.5, 3.0, and 2.5 floors.
<br>
<br>
- The distribution pattern indicates that the number of floors in houses varies based on whether they have a waterfront view or not. It seems that properties with a waterfront view are more likely to have 2.0 floors, while those without a waterfront view are commonly found with 1.0 floor.

### Year Built

In [29]:
#function to categorize year built
def categorize_year_decade(year):
    if year < 1900 :
        return '1800s'
    elif year < 1910 and year> 1900:
        return '1900s'
    elif year < 1920and year> 1910:
        return '1910s'
    elif year < 1930 and year> 1920:
        return '1920s'
    elif year < 1940 and year> 1930:
        return '1930s'
    elif year < 1950 and year> 1940:
        return '1940s'
    elif year < 1960 and year> 1950:
        return '1950s'
    elif year < 1970 and year> 1960:
        return '1960s'
    elif year < 1980 and year> 1970:
        return '1970s'
    elif year < 1990 and year> 1980:
        return '1980s'
    elif year < 2000 and year> 1990:
        return '1990s'
    else:
        return '2000s'

#function to create a new column 'year_decade'
df['year_decade'] = df['yr_built'].apply(categorize_year_decade)

# Group by 'year_decade'
df.groupby('year_decade')[['price', 'sqft_living', 'sqft_lot', 'grade', 'floors']].mean()

<>:7: SyntaxWarning: invalid decimal literal
<>:7: SyntaxWarning: invalid decimal literal
C:\Users\User\AppData\Local\Temp\ipykernel_24112\2872491017.py:7: SyntaxWarning: invalid decimal literal
  elif year < 1920and year> 1910:


price  sqft_living      sqft_lot     grade    floors
year_decade                                                              
1900s        641760.120072  1799.831541   6113.777778  7.186380  1.519713
1910s        570442.848214  1741.382440   9890.078869  6.919643  1.364583
1920s        606808.718464  1770.291590   7968.629799  7.137112  1.373400
1930s        602906.400000  1827.868182  16502.365909  7.104545  1.339773
1940s        433507.371269  1481.657960  10616.523010  6.660448  1.150498
1950s        479951.612171  1762.893733  12049.163942  7.169391  1.074932
1960s        456654.100746  1935.039386  13835.845771  7.340381  1.099710
1970s        498016.769159  2149.240130  19593.262889  7.693915  1.232002
1980s        531534.134182  2206.360921  21749.949070  8.015671  1.568071
1990s        586521.342023  2488.359228  24442.024505  8.380083  1.862617
2000s        593557.870586  2333.482960  13819.061253  8.088118  1.864139

- The analysis provides insights into how the decade of year built is associated with various attributes like price, living area size, lot size, grade, and floors.
- Generally, newer houses tend to have larger living areas, higher lot sizes, higher grades, and more floors.
- Houses from the 1990s and 2000s have higher average grades and living area sizes compared to earlier decades, which could reflect advancements in design and construction quality.
- While there are fluctuations, overall, the prices have increased over the years, with more recent decades showing higher mean prices.

### Year Renovated	

In [30]:
# Convert 'yr_renovated' to a categorical variable
df['renovated'] = df['yr_renovated'].apply(lambda x: 'Renovated' if x > 0 else 'Not Renovated')

In [35]:
df.groupby('renovated')[['price', 'sqft_living', 'sqft_lot', 'grade', 'floors']].mean().round(1).reset_index()

renovated     price  sqft_living  sqft_lot  grade  floors
0  Not Renovated  530448.0       2069.3   15039.6    7.7     1.5
1      Renovated  760628.8       2320.7   16633.6    7.7     1.5

- Renovated houses tend to have a significantly higher average price, suggesting that renovations contribute to increased property value.
- Renovated houses also tend to have larger average living areas and lot sizes compared to houses that were not renovated.
- While both renovated and not renovated houses have the same average grade.

In [36]:
df['yr_renovated'].value_counts(normalize=True).mul(100)

0       95.771064
2014     0.421043
2013     0.171193
2003     0.166566
2005     0.161940
          ...    
1951     0.004627
1959     0.004627
1948     0.004627
1954     0.004627
1944     0.004627
Name: yr_renovated, Length: 70, dtype: float64

- The majority of houses (95.77%) do not have specific renovation years available. This indicates that a significant portion of the dataset lacks renovation information.
- A small percentage of houses were renovated in specific years, with the highest being 0.42% in 2014, followed by 0.17% in 2013.
- The high percentage (95.77%) of houses without renovation year data suggests that either these houses were not renovated or the renovation year information is missing.
-  The years with higher percentages of renovations (e.g., 2014, 2013) maybe due to certain events (earthqueke, explosion...) or the number of houses that underwent renovation increased in recent years

## Price_per_sqft

In [40]:
#price_per_sqft function
def price_per_sqft(row):
    return row['price'] / row['sqft_living']

# Apply the price_per_sqft function
df['price_per_sqft'] = df.apply(price_per_sqft, axis=1)

In [41]:
df.head()

id             date     price  bedrooms  bathrooms  sqft_living  \
0  7129300520  20141013T000000  221900.0         3       1.00         1180   
1  6414100192  20141209T000000  538000.0         3       2.25         2570   
2  5631500400  20150225T000000  180000.0         2       1.00          770   
3  2487200875  20141209T000000  604000.0         4       3.00         1960   
4  1954400510  20150218T000000  510000.0         3       2.00         1680   

   sqft_lot  floors  waterfront  view  ...  zipcode      lat     long  \
0      5650     1.0           0     0  ...    98178  47.5112 -122.257   
1      7242     2.0           0     0  ...    98125  47.7210 -122.319   
2     10000     1.0           0     0  ...    98028  47.7379 -122.233   
3      5000     1.0           0     0  ...    98136  47.5208 -122.393   
4      8080     1.0           0     0  ...    98074  47.6168 -122.045   

   sqft_living15  sqft_lot15  grade_label  view_label  year_decade  \
0           1340        5650         good         bad        1950s   
1           1690        7639         good         bad        1950s   
2           2720        8062         good         bad        1930s   
3           1360        5000         good         bad        1960s   
4           1800        7503         good         bad        1980s   

       renovated  price_per_sqft  
0  Not Renovated      188.050847  
1      Renovated      209.338521  
2  Not Renovated      233.766234  
3  Not Renovated      308.163265  
4  Not Renovated      303.571429  

[5 rows x 26 columns]

In [44]:
#median price per square foot
median = df['price_per_sqft'].median()

#mean price per square foot for different renovation
renovated = df[df['renovated'] == 'Renovated']['price_per_sqft'].mean()
not_renovated = df[df['renovated'] == 'Not Renovated']['price_per_sqft'].mean()

In [45]:
print("Median Price Per Square Foot:", median)
print("Mean Price Per Square Foot (Renovated):", renovated)
print("Mean Price Per Square Foot (Not Renovated):", not_renovated)

Median Price Per Square Foot: 244.64454976303318
Mean Price Per Square Foot (Renovated): 319.3794498013898
Mean Price Per Square Foot (Not Renovated): 261.74650576477705


- Renovated houses, on average, have a higher price per square foot than non-renovated houses. This supports the assumption that renovations contribute to increased property values.

In [50]:
#mean price per square foot for different year decades
df.groupby('year_decade')['price_per_sqft'].mean()

year_decade
1900s    362.860901
1910s    334.932027
1920s    350.420726
1930s    321.681970
1940s    302.142132
1950s    273.348871
1960s    234.669819
1970s    230.138638
1980s    238.033631
1990s    228.507833
2000s    258.480065
Name: price_per_sqft, dtype: float64

- **Early 20th century:** Homes built in the early 20th century (1900s and 1910s) had a higher mean price per square foot than homes built in other decades. This suggests that there has been a historical trend of premium pricing for older homes.
- **Mid-20th century:** The mean price per square foot saw a decline during the mid-20th century (1940s and 1950s). This may have been due to changes in housing preferences or market dynamics during that period.
- **Recent years:** In more recent years (2000s), the mean price per square foot has shown an uptick. This suggests that there is renewed interest and increased value for properties built in the 21st century.

### Zip Code

In [56]:
#average price for each zip code
average_price_by_zip = df.groupby('zipcode')['price'].mean().round(1)

print(average_price_by_zip)

zipcode
98001     280804.7
98002     234284.0
98003     294111.3
98004    1356524.0
98005     810289.7
           ...    
98177     676419.4
98178     310612.8
98188     289078.3
98198     302896.7
98199     792187.9
Name: price, Length: 70, dtype: float64


- *Price variability:* The average price of homes varies a lot from zip code to zip code. This means that some neighborhoods are more expensive than others.
- *High-value zones:* There are some zip codes, like 98004 and 98199, where the average price of homes is much higher than in other zip codes. These zip codes may be in desirable areas with good schools, access to amenities, or other features that make them more attractive to buyers.
- *Market diversity:* The diversity of average prices suggests that there are a variety of housing markets to choose from.